ui/src/components/NavBar/NavBar.jsx

```diff
@@ -22,12 +22,12 @@ export default function NavBar({ isLoggedIn, handleLogout, viewProfile, setViewP
             }
             {
                 isLoggedIn && !viewProfile ?
-                <Link to={`/profile`} className="icon"><BsPersonCircle onClick={() => setViewProfile(true)}/></Link> : null
+                <NavBarIcon route={`/profile`} icon={"BsPersonCircle"} toggleViewProfile={toggleViewProfile}/> : null
             }
 
             {
                 isLoggedIn && viewProfile ?
-                <Link to={`/`} className="icon" ><BsFillHouseFill onClick={() => setViewProfile(false)}/></Link> : null
+                <NavBarIcon route={`/`} icon={"BsFillHouseFill"} toggleViewProfile={toggleViewProfile}/> : null
             }
```

Comment:
Oops! Sorry I should have clarified what I meant here - 

I think that NavBarIcon can take in `isLoggedin` and `viewProfile` as props, and then within the NavBar Icon file itself, you can run the logic to see which icon etc.. to render. That way <NavBarIcon> as a component is only called once in the NavBar. This is also a bit more readable in the code since it is more outright obvious that there is only ever 1 icon that is rendered from the NavBar.

Reply:
Ahhh ok that makes more sense! Updating now!